### PR TITLE
[Blocked]Add tests for checking lock provisioning on files through testing app

### PR DESCRIPTION
**Pull request by dpakach**
## Description

Added API acceptance tests for checking lock provisioning using the testing app

## Checklist:

- [ ] Latest changes are published according to [instructions in README.md](https://github.com/owncloud/testing#publish-latest-version-as-github-release)

### CODE REVIEW

File: tests/acceptance/features/bootstrap/TestingAppContext.php [hunk 1]
```diff
@@ -436,6 +441,137 @@ public function theResponseShouldHaveTheAppEnabledStatusOfApp() {
...
+	 *
+	 * @return void
+	 */
+	public function lockingIsDisabled() {
+		$res = $this->featureContext->setSystemConfig('filelocking.enabled', 'false', 'boolean');
+		PHPUnit_Framework_Assert::assertSame("System config value filelocking.enabled set to boolean false", \trim($res['stdOut']));
+	}
+
+	/**
+	 * @When the administrator checks lock provisioning status using the testing API
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/TestingAppContext.php]:
```suggestion
	 * @When the administrator checks the lock provisioning status using the testing API
```

File: tests/acceptance/features/bootstrap/TestingAppContext.php [hunk 2]
```diff
@@ -436,6 +441,137 @@ public function theResponseShouldHaveTheAppEnabledStatusOfApp() {
...
+			'GET',
+			$this->getBaseUrl("/lockprovisioning"),
+			[],
+			$this->featureContext->getOcsApiVersion()
+		);
+		$this->featureContext->setResponse($response);
+	}
+
+	/**
+	 * @When the administrator creates lock for file :path with type :type for user :user
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/TestingAppContext.php]:
```suggestion
	 * @When the administrator creates a lock for the file :path with the type :type for user :user
```

File: tests/acceptance/features/bootstrap/TestingAppContext.php [hunk 3]
```diff
@@ -436,6 +441,137 @@ public function theResponseShouldHaveTheAppEnabledStatusOfApp() {
...
+			'POST',
+			$this->getBaseUrl("/lockprovisioning/{$type}/{$user}"),
+			['path' => $path],
+			$this->featureContext->getOcsApiVersion()
+		);
+		$this->featureContext->setResponse($response);
+	}
+
+	/**
+	 * @Given the administrator has created lock for file :path with type :type for user :user
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/TestingAppContext.php]:
```suggestion
	 * @Given the administrator has created a lock for the file :path with the type :type for user :user
```

File: tests/acceptance/features/bootstrap/TestingAppContext.php [hunk 4]
```diff
@@ -436,6 +441,137 @@ public function theResponseShouldHaveTheAppEnabledStatusOfApp() {
...
+	 *
+	 * @return void
+	 */
+	public function theAdministratorHasCreatedLockForFileWithTypeForUser($path, $type, $user) {
+		$this->theAdministratorCreatesLockForFileWithTypeForUser($path, $type, $user);
+		PHPUnit_Framework_Assert::assertSame(200, $this->featureContext->getResponse()->getStatusCode());
+	}
+
+	/**
+	 * @When the administrator checks lock for file :path with type :type for user :user
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/TestingAppContext.php]:
```suggestion
	 * @When the administrator checks the lock for the file :path with the type :type for user :user
```

File: tests/acceptance/features/bootstrap/TestingAppContext.php [hunk 5]
```diff
@@ -436,6 +441,137 @@ public function theResponseShouldHaveTheAppEnabledStatusOfApp() {
...
+			'GET',
+			$this->getBaseUrl("/lockprovisioning/{$type}/{$user}?path={$path}"),
+			[],
+			$this->featureContext->getOcsApiVersion()
+		);
+		$this->featureContext->setResponse($response);
+	}
+
+	/**
+	 * @When the administrator deletes lock from file :path with type :type for user :user
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/TestingAppContext.php]:
```suggestion
	 * @When the administrator deletes the lock of the file :path with the type :type for user :user
```

File: tests/acceptance/features/bootstrap/TestingAppContext.php [hunk 6]
```diff
@@ -455,5 +591,6 @@ public function setUpScenario(BeforeScenarioScope $scope) {
 			$this->featureContext->getBaseUrl(),
 			$this->featureContext->getOcPath()
 		);
+		$this->initialLockEnabled = \trim($this->featureContext->getSystemConfig('filelocking.enabled')["stdOut"]) == 'true';
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/TestingAppContext.php]: make the line shorter
use `===`

File: tests/acceptance/features/bootstrap/TestingAppContext.php [hunk 7]
```diff
@@ -436,6 +441,137 @@ public function theResponseShouldHaveTheAppEnabledStatusOfApp() {
...
+	 *
+	 * @param string $path
+	 * @param string $type
+	 * @param string $user
+	 *
+	 * @return void
+	 */
+	public function theAdministratorHasCreatedLockForFileWithTypeForUser($path, $type, $user) {
+		$this->theAdministratorCreatesLockForFileWithTypeForUser($path, $type, $user);
+		PHPUnit_Framework_Assert::assertSame(200, $this->featureContext->getResponse()->getStatusCode());
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/TestingAppContext.php]: make the line shorter

File: tests/acceptance/features/bootstrap/TestingAppContext.php [hunk 8]
```diff
@@ -436,6 +441,137 @@ public function theResponseShouldHaveTheAppEnabledStatusOfApp() {
...
+	}
+
+	/**
+	 * @Given locking is disabled
+	 *
+	 * @return void
+	 */
+	public function lockingIsDisabled() {
+		$res = $this->featureContext->setSystemConfig('filelocking.enabled', 'false', 'boolean');
+		PHPUnit_Framework_Assert::assertSame("System config value filelocking.enabled set to boolean false", \trim($res['stdOut']));
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/TestingAppContext.php]: make the line shorter

File: tests/acceptance/features/bootstrap/TestingAppContext.php [hunk 9]
```diff
@@ -436,6 +441,146 @@ public function theResponseShouldHaveTheAppEnabledStatusOfApp() {
 		}
 	}
 
+	/**
+	 * @Given locking is enabled
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/TestingAppContext.php]: We try to use "has been" sentences for actions like this:'
```
Given locking has been enabled
```

The problem is that:
```
Given locking is enabled
Then locking is enabled
```
both look the same to Behat-Gherkin.

In English, the words could be an action or a test. So we try not to use ordinary "is" sentences. For actions we write:
```
Given something has been done
When the user does something
```
and for then tests we write like:
```
Then something should be set to "xyz"
```

File: tests/acceptance/features/bootstrap/TestingAppContext.php [hunk 10]
```diff
@@ -436,6 +441,146 @@ public function theResponseShouldHaveTheAppEnabledStatusOfApp() {
...
+	public function lockingIsEnabled() {
+		$res = $this->featureContext->setSystemConfig('filelocking.enabled', 'true', 'boolean');
+		PHPUnit_Framework_Assert::assertSame(
+			"System config value filelocking.enabled set to boolean true",
+			\trim($res['stdOut'])
+		);
+	}
+
+	/**
+	 * @Given locking is disabled
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/TestingAppContext.php]: Same here,
```
Given locking has been disabled
```

**Comment by dpakach** [comment]
Moving all changes to PR #52

**Pull request closed without merging**